Autogenerated settings turn off too many things in Lockdown Mode

brentfulgham opened the pull request:
#### fb16e152e4f6b4e8d4d20035e34034431d046e57
<pre>
Autogenerated settings turn off too many things in Lockdown Mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=251103">https://bugs.webkit.org/show_bug.cgi?id=251103</a>
&lt;rdar://104482701&gt;

Reviewed by Elliott Williams.

We decided that &apos;developer&apos; state should refer to features that are typically off,
but might be enabled by developers debugging a problem. A number of features that
are expected to be enabled at all times were improperly labeled with this state, causing
our Lockdown Mode logic to turn them off.

Instead, we should use &apos;stable&apos; for features that developers may wish to turn off for
A/B testing reasons, but are not intended to be off for most use cases.

This patch corrects these problems, and exposes two WebRTC flags as developer options that
can be enabled for experimentation, but should not be used by default.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:

Canonical link: <a href="https://commits.webkit.org/259331@main">https://commits.webkit.org/259331@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6e4a84222b08a5749fff2cefd57968ae007fe734

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104592 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13671 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37500 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113871 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/174094 "Found 1 new test failure: storage/indexeddb/modern/deleteindex-4-private.html (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108510 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14790 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4596 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96929 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112821 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110357 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11413 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94449 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38985 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/108069 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93269 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26061 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80649 "Passed tests") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/94568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7026 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27419 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/92483 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/4805 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7143 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3993 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/30063 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/103429 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13184 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46975 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/101169 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6435 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8932 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/25130 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -1467,7 +1467,7 @@ CookieConsentAPIEnabled:
 CookieEnabled:
   type: bool
-  status: developer
+  status: embedder
```
Why is this an "embedder" feature. There is a Safari setting for this, isn't there?

brentfulgham replied:
This was previously a WebPreference, without UI. I thought we might find it useful to toggle as a web developer, but since it's already exposed through Safari UI I moved it back to 'embedder' which acts like 'WebPreference'. I should not have ever made it 'developer'.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2560,7 +2560,7 @@ HighlightAPIEnabled:
 HyperlinkAuditingEnabled:
   type: bool
-  status: developer
+  status: embedder
```
Why isn't this "stable" for example? I don't understand what makes this an embedder feature.

brentfulgham replied:
This was previously a WebPreference, which is what we now call 'embedder'. I thought it might be useful to toggle for developers, but it's unclear that we have any value in doing so. This basically reverts to the working state we had previously.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -4135,7 +4135,7 @@ NeedsKeyboardEventDisambiguationQuirks:
 NeedsSiteSpecificQuirks:
   type: bool
-  status: developer
+  status: embedder
```
Isn't this something exposed to Web developers via Web Inspector? Does this qualify as an embedder feature?

brentfulgham replied:
Right. Since WebInspector already exposes it, we don't need to generate a 'developer' switch for it. I should not have marked it as 'developer', since WebInspector already provides a real UI for it.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -1951,7 +1951,7 @@ DisabledAdaptationsMetaTagEnabled:
 DisallowSyncXHRDuringPageDismissalEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
I considered `embedder`, but ended up with `stable` since it previously had an Experiment Feature toggle, and wasn't sure if people used it. I am not opposed to moving it to `shipping`.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2452,7 +2452,7 @@ GetUserMediaRequiresFocus:
 GoogleAntiFlickerOptimizationQuirkEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
Again, `shipping` would lose the UI toggle we used to have, which might be useful for debugging. `stable` retains the toggle.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -3088,7 +3088,7 @@ IsLoggedInAPIEnabled:
 IsNSURLSessionWebSocketEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
I think we should move to `shipping` for our next major Safari release, but not for a SU.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -3273,7 +3273,7 @@ LayoutViewportHeightExpansionFactor:
 LazyIframeLoadingEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
This just shipped this past major release, so a little soon to remove the toggle.

cdumez replied:
In fairness, I didn't know it would remove the UI toggle when I made my comment. The distinction between stable and shipping wasn't clear to me. This feature has shipped so I thought shipping would be the better state. I viewed "stable" as enabled but not shipped yet.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -3286,7 +3286,7 @@ LazyIframeLoadingEnabled:
 LazyImageLoadingEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
We should move to shipping for fall release, or perhaps even sooner. I'm not opposed to removing the toggle for this.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -4467,7 +4467,7 @@ PerformanceResourceTimingSensitivePropertiesEnabled:
 PermissionsAPIEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -4618,7 +4618,7 @@ PrivateClickMeasurementFraudPreventionEnabled:
 ProcessSwapOnCrossSiteNavigationEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5206,7 +5206,7 @@ ServiceWorkerNavigationPreloadEnabled:
 ServiceWorkersEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5235,7 +5235,7 @@ ServiceWorkersUserGestureEnabled:
 SharedWorkerEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
I support moving this to shipping.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5572,7 +5572,7 @@ SpeakerSelectionRequiresUserGesture:
 SpeechRecognitionEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
I support moving this to shipping.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5667,7 +5667,7 @@ StorageAPIEstimateEnabled:
 StorageAccessAPIEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
I wanted to retain the toggle for this one.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5869,7 +5869,7 @@ TextRecognitionInVideosEnabled:
 ThirdPartyIframeRedirectBlockingEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
I support moving this to shipping.

cdumez commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -6699,7 +6699,7 @@ WebInspectorEngineeringSettingsAllowed:
 WebLocksAPIEnabled:
   type: bool
-  status: developer
+  status: stable
```
Shouldn't this be "shipping" ?

brentfulgham replied:
This is also relatively new, so we should keep the toggle until next major release.

brentfulgham committed:
Autogenerated settings turn off too many things in Lockdown Mode
https://bugs.webkit.org/show_bug.cgi?id=251103
<rdar://104482701>

Reviewed by Elliott Williams.

We decided that 'developer' state should refer to features that are typically off,
but might be enabled by developers debugging a problem. A number of features that
are expected to be enabled at all times were improperly labeled with this state, causing
our Lockdown Mode logic to turn them off.

Instead, we should use 'stable' for features that developers may wish to turn off for
A/B testing reasons, but are not intended to be off for most use cases.

This patch corrects these problems, and exposes two WebRTC flags as developer options that
can be enabled for experimentation, but should not be used by default.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:

Canonical link: https://commits.webkit.org/259331@main
